pkg/snmp/traps/oid_resolver.go

```diff
@@ -0,0 +1,185 @@
+	trapOID = strings.TrimSuffix(NormalizeOID(trapOID), ".0")
+	trapData, ok := or.traps[trapOID]
+	if !ok {
+		return TrapMetadata{}, fmt.Errorf("trap OID %s is not defined", trapOID)
+	}
+	return trapData, nil
+}
+
+// GetVariableMetadata returns VariableMetadata for a given variableOID and trapOID.
+// The trapOID should not be needed in theory but the Datadog Agent allows to define multiple variable names for the
+// same OID as long as they are defined in different file. The trapOID is used to differentiate between these files.
+func (or *MultiFilesOIDResolver) GetVariableMetadata(trapOID string, varOID string) (VariableMetadata, error) {
+	trapOID = strings.TrimSuffix(NormalizeOID(trapOID), ".0")
+	varOID = strings.TrimSuffix(NormalizeOID(varOID), ".0")
+	trapData, ok := or.traps[trapOID]
+	if !ok {
+		return VariableMetadata{}, fmt.Errorf("trap OID %s is not defined", trapOID)
+	}
+	varData, ok := trapData.variableSpecPtr[varOID]
+	if !ok {
+		return VariableMetadata{}, fmt.Errorf("variable OID %s is not defined", varOID)
+	}
+	return varData, nil
+}
+
+func getSortedFileNames(files []fs.DirEntry) []string {
+	fileNames := make([]string, 0, len(files))
+	for _, file := range files {
+		if file.IsDir() {
+			log.Debugf("not loading traps data from path %s: file is directory", file.Name())
+			continue
+		}
+		fileNames = append(fileNames, file.Name())
+	}
+
+	// Sort files alphabetically but put first the one shipped with the agent
+	sort.Slice(fileNames, func(i, j int) bool {
+		fileNameI := strings.ToLower(fileNames[i])
+		fileNameJ := strings.ToLower(fileNames[j])
+		if strings.HasPrefix(fileNameI, ddTrapDBFileNamePrefix) {
+			return true
+		} else if strings.HasPrefix(fileNameJ, ddTrapDBFileNamePrefix) {
+			return false
+		}
+		return fileNameI < fileNameJ
+	})
+	return fileNames
+}
+
+func (or *MultiFilesOIDResolver) updateFromFile(filePath string) error {
+	var fileReader io.ReadCloser
+	fileReader, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer fileReader.Close()
+	if strings.HasSuffix(filePath, ".gz") {
+		filePath = strings.TrimSuffix(filePath, ".gz")
+		uncompressor, err := gzip.NewReader(fileReader)
+		if err != nil {
+			return fmt.Errorf("unable to uncompress gzip file %s", filePath)
+		}
+		defer uncompressor.Close()
+		fileReader = uncompressor
+	}
+	var unmarshalMethod unmarshaller = yaml.Unmarshal
+	if strings.HasSuffix(filePath, ".json") {
+		unmarshalMethod = json.Unmarshal
+	}
+	return or.updateFromReader(fileReader, unmarshalMethod)
+}
+
+func (or *MultiFilesOIDResolver) updateFromReader(reader io.Reader, unmarshalMethod unmarshaller) error {
+	fileContent, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return err
+	}
+	var trapData trapDBFileContent
+	err = unmarshalMethod(fileContent, &trapData)
+	if err != nil {
+		return err
+	}
+
+	return or.updateResolverWithData(trapData)
+}
+
+func (or *MultiFilesOIDResolver) updateResolverWithData(trapDB trapDBFileContent) error {
```

Comment:
Do we need to return an `error` ? It seems there is no code path leading to an error being returned.